### PR TITLE
Adding Docker-Compose

### LINKED ISSUE #17: Setup Docker/Podman
I think a Docker/Podman Compose like setup with one container for rails and another for postgres is ideal. Containerization is an interesting and very useful concept that has been coming up recently and quite a good learning experience to set this up. 

Honestly rails setup is kind of annoying and this helps us with deployment as well. 

[This](https://docs.docker.com/compose/rails/) and [this one too](https://www.digitalocean.com/community/tutorials/containerizing-a-ruby-on-rails-application-for-development-with-docker-compose) are good starter guides but I'm sure many others exist. 

This ties in with https://github.com/dsc-umass/project-matching/issues/16 as well which requires a Dockerfile.

### PR DESCRIPTION
**Pull request by abhinavtripathy**
Added docker-compose with rails and postgres containers. Added shell scripts to access both containers. Added Dockerfile with yarn support and added db_setup script. Added documentation to readme for building and running containers with common errors. This closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,71 @@ Project Matching is an open source project that helps teams/clubs help match the
...
+You can press ctrl/command + C on the terminal to stop the container.
+
+### Accessing the Rails Container
+
+To access the Rails Container:
+
+```
+sudo bash docker_shell.sh
+```
+If you want to access the rails console, then run the following comand after the previous one:
```
Comment (Saakshaat) [inline README.md:45]: This is definitely not a priority but could you fix the spelling on `command`?

File: config/database.yml [hunk 2]
```diff
@@ -17,9 +17,10 @@
 default: &default
   adapter: postgresql
   encoding: unicode
-  # For details on connection pooling, see Rails configuration guide
-  # https://guides.rubyonrails.org/configuring.html#database-pooling
-  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  pool: 5
+  username: projectmatch
+  password: projectmatchpass
```
Comment (Saakshaat) [inline config/database.yml:22]: Again, not a priority right now but we could consider using our container's environment variables to save and access credentials.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      context: .
+      dockerfile: docker/Dockerfile
+    depends_on:
+      - postgres
+    links:
+      - postgres
+    volumes:
+      - ./:/usr/src/app
+    ports:
+      - "3000:3000"
```
Comment (Saakshaat) [inline docker-compose.yml:21]: Silly EOL error that might return to haunt us later. 

Consider using a Ruby formatter in a `.gitattribute` file that will automatically handle this, along with other formatting issues, when the changes are pushed to our remote. 

For Python, I use a PEP-8 formatter called Black; I found [this for Ruby](https://github.com/ruby-formatter/rufo), maybe this'll be helpful in our case.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+RUN yarn install
+
+# Installs gems listed in the Gemfile.lock & Package.json
+RUN bundle install
+
+# Exposes the 3000 port to access it outside of the image
+EXPOSE 3000
+
+# Default command that runs when the container starts
+CMD ["bash", "docker/docker_set_up.sh"]
```
Comment (Saakshaat) [inline docker/Dockerfile:29]: EOL error again.

File: docker/docker_set_up.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+rm tmp/pids/server.pid
+yarn install --check-files
+bundle exec rails server -b 0.0.0.0
```
Comment (Saakshaat) [inline docker/docker_set_up.sh:3]: @abhinavtripathy any specific reason why we're starting the server on this particular port?

File: docker_shell.sh [hunk 6]
```diff
@@ -0,0 +1 @@
+docker-compose exec rails ${@:-/bin/bash}
```
Comment (Saakshaat) [inline docker_shell.sh:1]: EOL 💀

File: docker/docker_set_up.sh (hunk 5 above)
Comment (abhinavtripathy) [reply docker/docker_set_up.sh:3]: it references the port specified in docker-compose

File: docker/db_setup.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Database Setup
+bundle exec rake db:create 
+bundle exec rake db:migrate
```
Comment (harsh183) [inline docker/db_setup.sh:3]: Replace with  rake db:schema:load

**Comment by harsh183** [comment]
Looks great!!

**Pull request merged**